src/zenml/client.py

```diff
@@ -4538,8 +4541,9 @@ def create_service_connector(
                 "Python packages and ZenML integrations and try again."
             )
 
-        if not resource_type and len(connector.resource_types) == 1:
-            resource_type = connector.resource_types[0].resource_type
+        if not resource_type:
+            if len(connector.resource_types) == 1:
+                resource_type = connector.resource_types[0].resource_type
```

Comment:
Why was this changed? Old syntax seemed cleaner IMO

Reply:
probably the result of going through several stages to get this right. I'll change it back.